### PR TITLE
don't swallow configuration error messages

### PR DESCRIPTION
**Pull request by danielhochman**
## What was changed
When running 2.5.0, the server would silently stop (no logs or error messages) with exit code 0 in the event there was an error processing configuration (no config found, bad value, etc). I fixed this by wrapping the error from `config.Load` in `cli.Exit`.

I also added error handling for any uncaught errors from the run in case there are other instances of this problem.

## Checklist

1. How was this tested:
Unit

1. Any docs updates needed?
No

### CODE REVIEW

File: cmd/server/main.go [hunk 1]
```diff
@@ -42,7 +42,12 @@ import (
 // main entry point for the web server
 func main() {
 	app := buildCLI()
-	_ = app.Run(os.Args)
+	err := app.Run(os.Args)
+	if err != nil {
+		// An unhandled error was returned, wrap it and run it through the default exit code handler. Any errors
+		// that make it here should be caught further up the call stack and wrapped with cli.Exit and the proper exit code.
+		cli.HandleExitCoder(cli.Exit(fmt.Sprintf("Unexpected error encountered: %v.", err), 9))
```
Comment (feedmeapples) [inline cmd/server/main.go:49]: would rather leave just the error w/o the additional `Unexpected error..`

Comment (feedmeapples) [reply cmd/server/main.go:49]: is the addition to find errors easier in the logs?

Comment (danielhochman) [reply cmd/server/main.go:49]: the idea was that the errors should never make it to this point, they should be handled at the Action level.

this only logs if an error was returned without `cli.Exit`.

**Review by feedmeapples — APPROVED** [review]

**Comment by feedmeapples** [comment]
ty for the PR!

**Pull request merged**